src/curvefit/core/data.py

```diff
@@ -79,8 +93,7 @@ def __init__(self, df, col_t, col_obs, col_covs, col_group, obs_space,
         self.col_group = col_group
         self.obs_space = obs_space
         self.obs_se_func = obs_se_func
-
-        self.data_specs = DataSpecs(col_t, col_obs, col_covs, col_group, col_obs_se, obs_space, obs_se_func)
+        self.col_obs_se = col_obs_se
 
         self.df.sort_values([self.col_group, self.col_t], inplace=True)
```

Comment:
The df is getting sorted here but it's not getting sorted back when you call .get_df().

Reply:
Actually, I see that you access df through get_df() in other places where you rely on the fact that it's still sorted. Maybe we then just should rename it to _get_df() to emphasize that this method is for internal use and is **not**  returning the original data frame?

Reply:
@aksholokhov Ah I see. Good point -- we should implement. Keeping this open until we do.

Reply:
@aksholokhov changed both `Data` methods to internal: 4cc4dc6